File: app/src/main/java/com/github/mattiadellepiane/gnssraw/ui/main/tabs/SettingsFragment.java
package com.github.mattiadellepiane.gnssraw.ui.main.tabs;

import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.preference.EditTextPreference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;
import androidx.preference.SwitchPreference;

import com.github.mattiadellepiane.gnssraw.R;

public class SettingsFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.root_preferences, rootKey);

        setServerSwitchListener();

        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getContext());
        if(!preferences.getBoolean("server_enabled", false)){
            EditTextPreference serverEditText = findPreference("server_address");
            EditTextPreference serverPort = findPreference("server_port");
            serverEditText.setEnabled(false);
            serverPort.setEnabled(false);
        }
        if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S){
            SwitchPreference fullTracking = findPreference("full_tracking");
            fullTracking.setEnabled(true);
        }
    }

    private void setServerSwitchListener(){
        SwitchPreference serverSwitch = findPreference("server_enabled");
        if (serverSwitch != null) {
            serverSwitch.setOnPreferenceChangeListener(
                (preference, boolValue)->{
                    boolean value = (Boolean) boolValue;
                    EditTextPreference serverEditText = findPreference("server_address");
                    serverEditText.setEnabled(value);
                    EditTextPreference serverPort = findPreference("server_port");
                    serverPort.setEnabled(value);
                    return true;
                }
            );
        }
    }

}

File: app/src/main/java/com/github/mattiadellepiane/gnssraw/ui/main/tabs/SettingsFragment.kt
package com.github.mattiadellepiane.gnssraw.ui.main.tabs


import android.os.Bundle
import com.github.mattiadellepiane.gnssraw.R
import android.os.Build
import androidx.preference.*

class SettingsFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
        setPreferencesFromResource(R.xml.root_preferences, rootKey)
        setServerSwitchListener()
        val preferences = PreferenceManager.getDefaultSharedPreferences(context)
        if (!preferences.getBoolean("server_enabled", false)) {
            val serverEditText = findPreference<EditTextPreference>("server_address")
            val serverPort = findPreference<EditTextPreference>("server_port")
            serverEditText!!.isEnabled = false
            serverPort!!.isEnabled = false
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            val fullTracking = findPreference<SwitchPreference>("full_tracking")
            fullTracking!!.isEnabled = true
        }
    }

    private fun setServerSwitchListener() {
        val serverSwitch = findPreference<SwitchPreference>("server_enabled")
        if (serverSwitch != null) {
            serverSwitch.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { preference: Preference?, boolValue: Any ->
                val value = boolValue as Boolean
                val serverEditText = findPreference<EditTextPreference>("server_address")
                serverEditText!!.isEnabled = value
                val serverPort = findPreference<EditTextPreference>("server_port")
                serverPort!!.isEnabled = value
                true
            }
        }
    }
}